### ISSUE TITLE
Error encoding GeneralNames: TypeError: unorderable types: NoneType() >= int()

### ISSUE THREAD

**Issue by joernheissler** [wbond/asn1crypto#35]
Related to #33:

Directly above the erroring line (parser.py:235) there is "tag = tag". Would that be the issue here?

```
#!/usr/bin/env python3

from asn1crypto.x509 import GeneralNames

der = bytes.fromhex(
    '302E820F73757363657274652E676F622E7665A01B060560865E0202A0120C105249462D472D32303030343033362D30'
)

alt = GeneralNames.load(der)
alt.dump(force=True)
```

```
Traceback (most recent call last):
  File "alt.py", line 10, in <module>
    alt.dump(force=True)
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 3975, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 3842, in _set_contents
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 1107, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 3512, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 3081, in _set_contents
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 789, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 1485, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 534, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/core.py", line 524, in dump
  File "/home/joern/env/lib/python3.5/site-packages/asn1crypto-0.21.1-py3.5.egg/asn1crypto/parser.py", line 237, in _dump_header
TypeError: unorderable types: NoneType() >= int()
```

```
  0  46: SEQUENCE {
  2  15: . [2] 'suscerte.gob.ve'
 19  27: . [0] {
 21   5: . . OBJECT IDENTIFIER '2 16 862 2 2'
 28  18: . . [0] {
 30  16: . . . UTF8String 'RIF-G-20004036-0'
       : . . . }
       : . . }
       : . }
```

**Comment by wbond** [comment]
This seems to be an issue with `core.Any`:

```
  asn1crypto.x509.GeneralNames Object #4318967120
    Header: 0x302e
      constructed universal tag 16
    Data: 0x820f73757363657274652e676f622e7665a01b060560865e0202a0120c105249462d472d32303030343033362d30
    asn1crypto.x509.GeneralName Object #4319094736
      Data: 0x820f73757363657274652e676f622e7665
        asn1crypto.x509.DNSName Object #4319120144
          Header: 0x820f
            primitive context tag 2 (implicitly tagged)
          Data: 0x73757363657274652e676f622e7665
            Native: suscerte.gob.ve
    asn1crypto.x509.GeneralName Object #4319120336
      Data: 0xa01b060560865e0202a0120c105249462d472d32303030343033362d30
        asn1crypto.x509.AnotherName Object #4319267664
          Header: 0xa01b
            constructed context tag 0 (implicitly tagged)
          Data: 0x060560865e0202a0120c105249462d472d32303030343033362d30
            Field "type_id"
              asn1crypto.core.ObjectIdentifier Object #4319265616
                Header: 0x0605
                  primitive universal tag 6
                Data: 0x60865e0202
                  Native: 2.16.862.2.2
            Field "value"
              asn1crypto.core.Any Object #4319267472
                Header: 0xa012
                  context tag 0 (explicitly tagged)
                Data: 0x0c105249462d472d32303030343033362d30
                  asn1crypto.core.UTF8String Object #4319267408
                    Header: 0xa0120c10
                      context tag None (explicitly tagged)
                        primitive universal 12
                    Data: 0x5249462d472d32303030343033362d30
                      Native: RIF-G-20004036-0
```

**Comment by wbond** [comment]
Fixed by 02c3d992f28335ff13fe8620d254a77e86def6c6
